packages/aws-rfdk/lib/deadline/lib/repository.ts

```diff
@@ -271,6 +275,34 @@ export interface RepositorySecurityGroupsOptions {
   readonly installer?: ISecurityGroup;
 }
 
+/**
+ * Settings used by Deadline Secrets Management, a feature introduced in Deadline 10.1.10 for securely managing storage
+ * and access of Secrets for your render farm.
+ * More details at:
+ * https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/secrets-management/deadline-secrets-management.html
+ * Using Secrets Management requires TLS to be enabled between the RenderQueue and its clients. If this feature is enabled, the
+ * `externalTLS` on the `RenderQueueTrafficEncryptionProps` interface on the RenderQueue cannot be disabled.
+ */
+export interface SecretsManagementProps {
+  /**
+   * Whether or not to enable the Secrets Management feature.
+   * @default true
+   */
+  readonly enabled?: boolean;
+  /**
+   * A Secret containing the username and password to use for the admin role.
+   * The contents of this secret must be a JSON document with the keys "username" and "password". ex:
+   *     {
+   *         "username": <admin user name>,
+   *         "password": <admin user password>,
+   *     }
+   * Password should contain at least one lowercase letter, one uppercase letter, one symbol and one number.
```

Comment:
I just realized we have no easy way to do validation on the password that the user supplies. This means that if they supply a bad password, we won't find out until the CDK deployment gets to the point that the repository installer is running. This isn't ideal, but I don't see a better way to do it.

We could have a Custom Resource that runs a Lambda to do validation on it, but that also would happen at deploy time. At least if we did that, it would be more obvious to the user why the deployment failed. As it stands now, if the deployment fails, they would have to go check the logs in CloudWatch.

Maybe we should add a note here to say something like:
If the password requirements aren't met, the repository installer will fail during the deployment. We recommend leaving this blank so that a random password will be generated that meets the requirements.

Reply:
I just noticed that the password must also be at least 8 characters long. We should add that to the requirement.